### PR TITLE
Add install_info file

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Creates `install_info` file with information about the install method used (overwrites any other existing file).

### Describe your test plan

I added some spec tests. Unfortunately in the test environment one can't get the puppet version, so that has to be tested manually:

- Checkout this branch and run `pdk build` to create a .tar.gz of this version
- Install it using `sudo puppet module install <path to tar>`
- Apply it following the repo README
- Check that the `install_info` file was created with correct content and permissions

### CODE REVIEW

File: spec/classes/datadog_agent_spec.rb [hunk 1]
```diff
@@ -1592,17 +1592,37 @@
...
         it { is_expected.to contain_file(config_dir + '/conf.d').with_ensure('directory') }
 
         # Agent 5 files
         it { is_expected.not_to contain_file('/etc/dd-agent') }
         it { is_expected.not_to contain_concat('/etc/dd-agent/datadog.conf') }
         it { is_expected.not_to contain_file('/etc/dd-agent/conf.d').with_ensure('directory') }
 
+        describe 'install_info check' do
+          it {
+            is_expected.to contain_file(install_info_file).with(
```
Comment (kbogtob) [inline spec/classes/datadog_agent_spec.rb:1610]: Why not parse it as YAML in Ruby and check the keys' values are correct?

File: spec/classes/datadog_agent_spec.rb [hunk 2]
```diff
@@ -1592,17 +1592,37 @@
 
         config_dir = WINDOWS_OS.include?(operatingsystem) ? 'C:/ProgramData/Datadog' : '/etc/datadog-agent'
         config_yaml_file = config_dir + '/datadog.yaml'
+        install_info_file = config_dir + '/install_info'
```
Comment (kbogtob) [inline spec/classes/datadog_agent_spec.rb:1595]: Could you use `File.join`?

File: manifests/init.pp [hunk 3]
```diff
@@ -714,6 +725,14 @@
         require => File['C:/ProgramData/Datadog'],
       }
 
+      file { 'C:/ProgramData/Datadog/install_info':
+        owner   => $dd_user,
+        group   => $dd_group,
+        mode    => '0660',
```
Comment (kbogtob) [inline manifests/init.pp:731]: Why different permissions? (640 or 660)

File: spec/classes/datadog_agent_spec.rb (hunk 1 above)
Comment (mx-psi) [reply spec/classes/datadog_agent_spec.rb:1610]: Sure! I was trying to follow the style of the existing tests, I guess we should refactor those as well at some point 🤔

File: manifests/init.pp (hunk 3 above)
Comment (mx-psi) [reply manifests/init.pp:731]: I am not sure what the reason is for it but `datadog.yaml` has those permissions on Windows, so I guessed `install_info` should have the same. Should I change them on `datadog.yaml` too?

Comment (mx-psi) [reply manifests/init.pp:731]: Seems like it is used to give the administrator group access to the file (see #571)

**Pull request merged**